larray/viewer.py

```diff
@@ -2293,12 +2296,17 @@ def _add_arrays(self, arrays):
         for k, v in arrays.items():
             self.data[k] = v
             self.add_list_item(k)
+        if qtconsole_available:
+            self.kernel.shell.push(dict(arrays))
 
     def _clear_arrays(self):
         arrays = [k for k, v in self.data.items() if self._display_in_grid(k, v)]
         for name in arrays:
             del self.data[name]
             self.delete_list_item(name)
+        if qtconsole_available:
+            for name in arrays:
+                self.kernel.shell.del_var(name)
```

Comment:
I wonder if it wouldn't be possible to refactor delete_current_item, _add_arrays, update_mapping, etc. to reduce duplication of code.